### PR TITLE
Static analysis fixes

### PR DESCRIPTION
**Pull request by littleaj**
* fix some potential NPEs
* fix String equality check
* fix InterruptedException handling
* remove redundant conditional/logging

### CODE REVIEW

File: core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulse.java [hunk 1]
```diff
@@ -149,13 +149,14 @@ public synchronized void stop(long timeout, TimeUnit timeUnit) {
         try {
```
Comment (trask) [inline core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulse.java]: Btw, looking at this, I noticed above code calling `stop()` never throws any exception, so the above catch blocks are unnecessary.

File: core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulse.java [hunk 2]
```diff
@@ -149,13 +149,14 @@ public synchronized void stop(long timeout, TimeUnit timeUnit) {
...
             thread.join();
         } catch (InterruptedException e) {
-            InternalLogger.INSTANCE.trace("Stack trace generated is %s", ExceptionUtils.getStackTrace(e));
+            Thread.currentThread().interrupt();
```
Comment (trask) [inline core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulse.java]: Maybe better to `interrupt()` both threads first, then try to `join()` them, and if either throws `InterruptedException`, then re-`interrupt()` and `return` from method (without setting `initialized` to `false` at the bottom?)

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ActiveTransmissionLoader.java [hunk 3]
```diff
@@ -152,6 +153,7 @@ public synchronized boolean load(boolean waitForThreadsToStart) {
             return true;
```
[the remark this answers is not in the document]
Comment (littleaj) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ActiveTransmissionLoader.java]: Yeah, this does look a bit odd.

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ActiveTransmissionLoader.java [hunk 4]
```diff
@@ -168,6 +170,7 @@ public void stop(long timeout, TimeUnit timeUnit) {
                 thread.join();
             } catch (InterruptedException e) {
                 InternalLogger.INSTANCE.error("Interrupted during join of active transmission loader, exception: %s", e.toString());
+                Thread.currentThread().interrupt();
```
[the remark this answers is not in the document]
Comment (littleaj) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ActiveTransmissionLoader.java]: makes sense

File: core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulse.java (hunk 2 above)
Comment (littleaj) [reply core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulse.java]: I think `initialized` should still be set to false after interrupting the worker threads. Both workers have already had signals to stop their runnables.

File: core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ActiveTransmissionLoader.java [hunk 5]
```diff
@@ -95,6 +95,7 @@ public void run() {
                         barrier.await();
```
[the remark this answers is not in the document]
Comment (littleaj) [reply core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/ActiveTransmissionLoader.java]: agreed.

**Pull request merged**